srtcore/core.cpp

```diff
@@ -12289,11 +12289,17 @@ int CUDTGroup::sendBroadcast(const char* buf, int len, SRT_MSGCTRL& w_mc)
 int CUDTGroup::getGroupData(SRT_SOCKGROUPDATA* pdata, size_t* psize)
 {
     CGuard gl (m_GroupLock);
-
-    size_t size = *psize;
+    
+    SRT_ASSERT(psize != NULL);
+    const size_t size = *psize;
     // Rewrite correct size
     *psize = m_Group.size();
 
+    if (!pdata)
+    {
+        return 0;
```

Comment:
0 is defined as "successful" return and it means that the group data have been retrieved and the array filled. Here it's not possible.

Reply:
If `pdata=NULL` then the intention of the function is to get the number of socket group members, and the function succeeds in this case.

Reply:
I'm thinking about the proper recognition possibilities. Maybe then also when the size is exceeded, the error code should be also properly set. I was thinking about MN_XSIZE. The actual meaning of this error could be extended.